go-controller/pkg/ovn/controller/unidling/unidle.go

```diff
@@ -0,0 +1,138 @@
+package unidling
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+
+	"k8s.io/klog/v2"
+)
+
+// unidlinController checks periodically the OVN events db
+// and generates a Kubernetes NeedPods events with the Service
+// associated to the VIP
+type unidlingController struct {
+	eventRecorder record.EventRecorder
+	// Map of load balancers to service namespace
+	serviceVIPToName     map[ServiceVIPKey]types.NamespacedName
+	serviceVIPToNameLock sync.Mutex
```

Comment:
we should use a RW mutex here. The reason is because there is going to be tons of get requests because of how OVN generates an event for every packet. So there are going to be way more reads than writes so we should use read lock. We could opt for a sync map instead if we dont plan on iterating over this map (not thread safe with sync maps).

Reply:
the reads are serialised, there is no difference on using a RWmutex AFAIK.
I will be conservative here and keep it as is, there is no clear improvement